crates/supply/src/lib.rs

```diff
@@ -163,25 +167,28 @@ impl<T: Config> Pallet<T> {
         T::SupplyPalletId::get().into_account_truncating()
     }
 
-    pub(crate) fn begin_block(height: T::BlockNumber) {
+    pub(crate) fn begin_block(height: T::BlockNumber) -> DispatchResult {
         // ignore if uninitialized or not start height
         if let Some(start_height) = <StartHeight<T>>::get().filter(|&start_height| height == start_height) {
             let end_height = start_height + T::InflationPeriod::get();
             <StartHeight<T>>::put(end_height);
 
             let total_supply = T::Currency::total_issuance();
-            let total_supply_as_fixed = T::UnsignedFixedPoint::checked_from_integer(total_supply).unwrap();
+            let total_supply_as_fixed =
+                T::UnsignedFixedPoint::checked_from_integer(total_supply).ok_or(ArithmeticError::Overflow)?;
             let total_inflation = total_supply_as_fixed
                 .saturating_mul(<Inflation<T>>::get())
                 .truncate_to_inner()
-                .unwrap_or(Zero::zero());
+                .unwrap_or_default();
```

Comment:
I think this can be better expressed as `<Inflation<T>>::get().checked_mul_int(total_supply)`. That saves us from having to to the fixedpoint conversions, and if we anyway return result, the checked_mul_int error can also be propagated. I think it's also less likely to overflow